connection.go

```diff
@@ -326,6 +326,29 @@ func (cn *connection) PeerHasPiece(piece pieceIndex) bool {
 	return cn.peerSentHaveAll || cn.peerPieces.Contains(bitmap.BitIndex(piece))
 }
 
+func (cn *connection) PeerBitfield() []bool {
```

Comment:
The allocation is unnecessary, perhaps make it unexported to allow changes in the future if it becomes a performance bottleneck?